src/ckeditor.jsx

```diff
@@ -5,24 +5,67 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
+import EditorWatchdog from '@ckeditor/ckeditor5-watchdog/src/editorwatchdog';
+import uid from '@ckeditor/ckeditor5-utils/src/uid';
+import { ContextWatchdogContext } from './context.jsx';
+import ContextWatchdog from '@ckeditor/ckeditor5-watchdog/src/contextwatchdog';
 
 export default class CKEditor extends React.Component {
 	constructor( props ) {
 		super( props );
 
 		// After mounting the editor, the variable will contain a reference to the created editor.
 		// @see: https://ckeditor.com/docs/ckeditor5/latest/api/module_core_editor_editor-Editor.html
-		this.editor = null;
 		this.domContainer = React.createRef();
+
+		/**
+		 * An instance of EditorWatchdog if the context watchdog was not passed via React's context. `null` otherwise.
+		 */
+		this.watchdog = null;
+
+		/**
+		 * An instance of ContextWatchdog passed from the React's context or `null` otherwise.
+		 */
+		this.contextWatchdog = null;
+
+		/**
+		 * A unique id for an editor component.
+		 */
+		this._id = uid();
+	}
+
+	get editor() {
+		if ( this.watchdog ) {
+			return this.watchdog.editor;
+		}
+
+		if ( this.contextWatchdog ) {
+			// TODO - try/catch should not be necessary as `getItem` could return `null` instead of throwing errors.
+			try {
+				return this.contextWatchdog.getItem( this._id );
+			} catch ( err ) {
+				return null;
+			}
+		}
+
+		return null;
 	}
 
-	// This component should never be updated by React itself.
+	// This component should almost never be updated by React itself.
 	shouldComponentUpdate( nextProps ) {
 		if ( !this.editor ) {
 			return false;
 		}
 
-		if ( this._shouldUpdateContent( nextProps ) ) {
+		// Only when the editor component changes the whole structure should be restarted.
+		if ( nextProps.id !== this.props.id ) {
+			this._destroyEditor();
+			this._initializeEditor();
+
+			return true;
+		}
```

Comment:
I understand what we want to do here. But I am unsure should the component be updated by React? I guess, manually destroying and creating a new editor should be fine.

Also:
- `Editor#destroy()` returns a promise. Shouldn't we wait for it?
- Same with `Editor#create()`.
- I'd rename `props.id` to something other. Maybe `editorId`. It can be confusing to set the `id` attribute:

```jsx
<CKEditor [...] id="my-ckeditor-element">
```

As for me, I'd expect that the editor container will be marked with the specified identifier.